index.md

```diff
@@ -1,71 +1,108 @@
-# Programming Essentials
+---
+title: "Table of Contents"
+date: "2023-10-02"
+author: "Andrei Guevorkian"
+---
 
-Purpose of section: 
-- Teach readers how to think about programming
-- Teach readers how to approach a problem
-- Ask questions to make readers think about programming
+## Computer Fundamentals
+
+### [Basics of Computers](./computer-fundamentals/basics)
+
+- Tracing the evolution of computing devices.
+- The transformation of user interaction over time.
+- Distinguishing features that define a "computer".
+- The importance and role of binary in computation.
+
+### [Software and Hardware](./computer-fundamentals/software-and-hardware)
+
+- Delving into the fundamental parts of a computer: Input, Processing Unit, Memory & Storage, and Output.
+- Distinguishing between tangible hardware and intangible software.
+- Understanding Operating Systems: Bridging the gap between software and hardware.
+- The File System: Drawing parallels with a school locker analogy.
+- Exploring the intricacies of File Naming, File Path, and Working Directory.
+
+## Programming Essentials
+
+### [Introduction](./programming-fundamentals/introduction)
 
-## Introduction
 - What is programming and why is it important?
 - Real-life examples of how programming impacts our daily life.
 - The mindset of a programmer: Logical, creative, and persistent.
 
-## Chapter 1: Understanding and Breaking Down Problems
+### [Chapter 1: Understanding and Breaking Down Problems](./programming-fundamentals/chapter1)
+
 - The importance of clearly defining a problem.
 - Strategies to understand complex problems:
-    - Breaking them down into smaller parts: The Divide and Conquer approach.
-    - Visualizing with diagrams or flowcharts.
+  - Breaking them down into smaller parts: The Divide and Conquer approach.
+  - Visualizing with diagrams or flowcharts.
 - Activity: Solving a larger problem by breaking it down.
 
-## Chapter 2: Algorithms and Flowcharts
+### [Chapter 2: Algorithms and Flowcharts](./programming-fundamentals/chapter2)
+
 - What is an algorithm?
 - Introduction to flowcharts and their symbols.
 - How to convert a problem into a flowchart.
 - Activity: Designing a flowchart for a simple daily task.
 
-## Chapter 3: Logic and Decision Making
+### [Chapter 3: Logic and Decision Making](./programming-fundamentals/chapter3)
```

Comment:
If the corresponding Markdown file does not yet exist, I don't think we should link to it.

Reply:
About the render, should render the best result when the author follow the rules.
For that not follow the rules, just output normal result, but can not render to crash.